### DISCUSSION TITLE
Gem.install_extension_in_lib and pre-compiled gems

### DISCUSSION THREAD

**Discussion by MSP-Greg** [ruby/rubygems#6265] [General]
When `Gem.install_extension_in_lib` is switched to `false` (RG 4?), gem extension files will only be placed in the `extensions` folder.

How will this work with pre-compiled gems?

**Comment by simi** [comment]
Precompiled gems doesn't install extensions. Extensions are just extracted since they are part of the gem package. I wasn't looking into code, but I would assume, that config is not related for pre-compiled gems.

Or is there already any case to share breaking the loading of precompiled gem using this option?

**Reply by MSP-Greg** [reply]
I was wondering about editing the first post, as there are two operations.

The first, which where 'installed extensions' are placed.

The second, which is what locations are searched for the extension when it is loaded/activated (after the gem is installed).

If RG4 will continue to search in both locations, pre-compiled gems will work fine.

But, if a gem is installed that compiles the extension (not a pre-compiled gem), and if loads the extension with `require_relative`, it won't find the extension in the `extensions` folder?

Currently, some platforms place extensions in a platform specific folder, so gems should be loading extensions with `require`, not `require_relative`...

**Reply by simi** [reply]
Extension install is the only place being affected by `install_extension_in_lib`.  It doesn't affect any load path.

**Comment by deivid-rodriguez** [comment]
While precompiled gems don't compile extensions, gems that do compile extensions will indeed break if they rely on the extension being loaded relatively from files in `lib` directory. @MSP-Greg How did you find about this? Do you have examples of extensions that will break? I found ox and proposed a fix for it [here](https://github.com/ohler55/ox/pull/383).
